### ISSUE TITLE
Detect boolean literals in logical expressions

### ISSUE THREAD

**Issue by bvanjoi** [rust-lang/rust#127801]
### Code

```Rust
fn main() {
  let _ = false || true; // expect to throw a warning that it will always return `true`.
  let _ = true || false;
  let _ = false && true; // expect to throw a warning that it will always return `false`.
  let _ = true && false;
}
```

### Current output

```Shell
compile succeeded, and no warnings were reported.
```

### Rust Version

```Shell
rustc 1.81.0-nightly (24d2ac0b5 2024-07-15)
binary: rustc
commit-hash: 24d2ac0b56fcbde13d827745f66e73efb1e17156
commit-date: 2024-07-15
host: aarch64-apple-darwin
release: 1.81.0-nightly
LLVM version: 18.1.7
```

**Comment by eggyal** [comment]
How often do boolean expressions with multiple literal operands actually arise in practice?  Feels rather niche.

**Comment by ShE3py** [comment]
Clippy already lints those four cases. `false && cond` & `true || cond` are often used for quick debugging, so not warning what the user intended to do is a nice thing imo.

**Comment by tgross35** [comment]
As noted, [`clippy::nonminimal_bool`](https://rust-lang.github.io/rust-clippy/master/index.html#/nonminimal_bool) covers these. In theory they could be uplifted to regular lints, but this feels more like a cleanliness lint rather than correctness.
